crates/node/builder/src/rpc.rs

```diff
@@ -461,13 +509,152 @@ where
+    /// starting the auth server.
+    pub async fn launch_rpc_server<F>(
+        self,
+        ctx: AddOnsContext<'_, N>,
+        ext: F,
+    ) -> eyre::Result<RpcServerOnlyHandle<N, EthB::EthApi>>
+    where
+        F: FnOnce(
+            &mut TransportRpcModules,
+            &mut AuthRpcModule,
+            &mut RpcRegistry<N, EthB::EthApi>,
+        ) -> eyre::Result<()>,
+    {
+        let setup_ctx = self.setup_rpc_components(ctx, ext).await?;
+        let RpcSetupContext {
+            node,
+            config,
+            mut modules,
+            mut auth_module,
+            auth_config: _,
+            mut registry,
+            on_rpc_started,
+            engine_events,
+            beacon_engine_handle,
+        } = setup_ctx;
+
+        let server_config = config.rpc.rpc_server_config();
+        let rpc_server_handle = Self::launch_rpc_server_internal(server_config, &modules).await?;
+
+        let handles =
+            RethRpcServerHandles { rpc: rpc_server_handle.clone(), auth: AuthServerHandle::noop() };
+        Self::finalize_rpc_setup(
+            &mut registry,
+            &mut modules,
+            &mut auth_module,
+            &node,
+            config,
+            on_rpc_started,
+            handles,
+        )?;
+
+        Ok(RpcServerOnlyHandle {
+            rpc_server_handle,
+            rpc_registry: registry,
+            engine_events,
+            beacon_engine_handle,
+        })
+    }
+
+    /// Launches only the authenticated Engine API server, without the regular RPC server.
+    ///
+    /// This is useful for specialized setups where you only need the Engine API functionality
+    /// for consensus layer communication.
+    pub async fn launch_auth_server<F>(
+        self,
+        ctx: AddOnsContext<'_, N>,
+        ext: F,
+    ) -> eyre::Result<AuthServerOnlyHandle<N, EthB::EthApi>>
+    where
+        F: FnOnce(
+            &mut TransportRpcModules,
+            &mut AuthRpcModule,
+            &mut RpcRegistry<N, EthB::EthApi>,
+        ) -> eyre::Result<()>,
+    {
+        let setup_ctx = self.setup_rpc_components(ctx, ext).await?;
+
+        let auth_server_handle =
+            Self::launch_auth_server_internal(setup_ctx.auth_module.clone(), setup_ctx.auth_config)
+                .await?;
+
+        Ok(AuthServerOnlyHandle {
+            auth_server_handle,
+            rpc_registry: setup_ctx.registry,
+            engine_events: setup_ctx.engine_events,
+            beacon_engine_handle: setup_ctx.beacon_engine_handle,
+        })
+    }
+
     /// Launches the RPC servers with the given context and an additional hook for extending
     /// modules.
     pub async fn launch_add_ons_with<F>(
         self,
         ctx: AddOnsContext<'_, N>,
         ext: F,
     ) -> eyre::Result<RpcHandle<N, EthB::EthApi>>
+    where
+        F: FnOnce(
+            &mut TransportRpcModules,
+            &mut AuthRpcModule,
+            &mut RpcRegistry<N, EthB::EthApi>,
```

Comment:
I also think we could introduce a helper container type for this so we can add changes to this more easily

Reply:
good point, done